packages/test-sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/util/TestAppInitializer.kt

```diff
@@ -470,10 +470,10 @@ object TestAppInitializer {
         source =
         """
         exports = ({mail, id}) => {
-            // Auth function will fail for emails with a domain different to @androidtest.realm.io
+            // Auth function will fail for emails with a domain different to @10gen.com
             // or with id lower than 666
-            if (!new RegExp("@androidtest.realm.io${'$'}").test(mail) || id < 666) {
-                return 0;
+            if (!new RegExp("@10gen.com${'$'}").test(mail) || id < 666) {
+                throw new Error(`Authentication failed`);
```

Comment:
I suspect we need to test both `return 0` and `throw` here.

We should verify in tests that proper AuthException errors are being thrown in both cases. I would expect  `InvalidCredentialsException` if the credentials fail or `AuthException` if an exception is thrown.

You can see in our error mapping logic that we are doing some string comparisons to be able to throw `InvalidCredentialsException` for some of the other credential types.

Reply:
The custom function should return the user id as a `String` or `Object` or throw an exception in case of failure. If we return 0 core would fail to parse the user id. 

It generates an AuthFailure too, but with an error message stating the parsing failure.